ca_cdk_constructs/edge_services/waf_rule_templates.py

```diff
@@ -179,3 +179,122 @@ def ip_rule_property(
             )
         ),
     )
+
+def restricted_uri_string_property(
+    scope,
+    name: str,
+    priority: int,
+    restricted_uri_string: str,
+    allowed_addresses: dict[str, list[str]] = {},
+    count_only: bool = False,
+    cloud_watch_metrics_enabled: bool = False,
+) -> waf.CfnWebACL.RuleProperty:
+    """A wrapper that returns an `aws_wafv2.CfnWebACL.RuleProperty` object to be used in a list and passed to the CfnWebACL instance.
+
+    :param scope: `self`, scope in which this resource is defined.
+    :param name: The name of the rule. You can't change the name of a Rule after you create it.
+    :param priority: If you define more than one Rule in a WebACL , AWS WAF evaluates each request against the Rules in order based on the value of Priority . AWS WAF processes rules with lower priority first. The priorities don't need to be consecutive, but they must all be different.
+    :param restricted_uri_string: Access to any URL containing this string will be restricted to the IP addresses in allowed_addresses
+    :param allowed_addresses: A dictionary of strings that specifies zero or more IP addresses or blocks of IP addresses for "IPV4" and "IPV6". Defaults to {}. All addresses must be specified using Classless Inter-Domain Routing (CIDR) notation.
+    :param count_only: Set to True to only count and not BLOCK on matching requests, defaults to False.
+    :param cloud_watch_metrics_enabled: Set to True to enable logging via Cloudwatch, defaults to Fasle.
+    :return: aws_cdk.aws_wafv2.CfnWebACL.RuleProperty.
+
+    :example:
+    ```
+     restricted_uri_string_property(
+        scope,
+        name=f"{scope.stack_name}AllowToAccessUriString-helptoclaim",
+        priority=0,
+        restricted_uri_string="helptoclaim"
+        allowed_addresses={"IPV4": ["1.1.1.1/32"], "IPV6": ["2a00:1d40:11a5::111"]},
+        count_only=True,
+    )
+    """
+
+    # Count must be set to {} or None.
+    if count_only:
+        count_only = {}
+    else:
+        count_only = None
```

Comment:
can get rid of that as `count_only` already default to `False` in the constructor

Reply:
@zhelyan I've checked this, and if you remove the `else:` clause it breaks the code, and you get a whole raft of jsii errors, which disappear as soon as you reinstate it. So I propose we keep it because we know that this works. So in this context, clearly `False` and `None` are not equivalent.